plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/impl/BaseFlightProducer.java

```diff
@@ -0,0 +1,283 @@
+                StreamProducer.BatchedJob<VectorSchemaRoot> batchedJob = producer.createJob(allocator);
+                if (context.isCancelled()) {
+                    handleCancellation(batchedJob, listener);
+                    return;
+                }
+                processStream(streamProducerHolder, batchedJob, context, listener);
+            }
+        } catch (Exception e) {
+            logger.error("Unexpected error during stream processing for ticket: " + streamTicket, e);
+            listener.error(
+                CallStatus.INTERNAL.withDescription("Internal server error: " + e.getMessage()).withCause(e).toRuntimeException()
+            );
+        }
+    }
+
+    /**
+     * Retrieves FlightInfo for a given descriptor, handling both local and remote cases.
+     * The descriptor's command is expected to contain a serialized StreamTicket.
+     *
+     * @param context The call context
+     * @param descriptor The flight descriptor containing stream information
+     * @return FlightInfo for the requested stream
+     * @throws RuntimeException if the requested info cannot be retrieved
+     */
+    @Override
+    public FlightInfo getFlightInfo(CallContext context, FlightDescriptor descriptor) {
+        StreamTicket streamTicket;
+        try {
+            streamTicket = streamManager.getStreamTicketFactory().fromBytes(descriptor.getCommand());
+        } catch (Exception e) {
+            logger.debug("Failed to parse flight descriptor command into StreamTicket", e);
+            throw CallStatus.INVALID_ARGUMENT.withDescription("Invalid descriptor format: " + e.getMessage())
+                .withCause(e)
+                .toRuntimeException();
+        }
+
+        if (streamTicket.getNodeId().equals(flightClientManager.getLocalNodeId())) {
+            return getLocalFlightInfo(streamTicket, descriptor);
+        } else {
+            return getRemoteFlightInfo(streamTicket, descriptor);
+        }
+    }
+
+    private FlightStreamManager.StreamProducerHolder acquireStreamProducer(
+        StreamTicket streamTicket,
+        Ticket ticket,
+        ServerStreamListener listener
+    ) {
+        if (streamTicket.getNodeId().equals(flightClientManager.getLocalNodeId())) {
+            return streamManager.removeStreamProducer(streamTicket).orElse(null);
+        }
+        Optional<FlightClient> remoteClient = flightClientManager.getFlightClient(streamTicket.getNodeId());
+        if (remoteClient.isPresent()) {
+            try {
+                StreamProducer<VectorSchemaRoot, BufferAllocator> proxyProvider = createProxyProducer(remoteClient.get(), ticket, listener);
+                if (proxyProvider != null) {
+                    return FlightStreamManager.StreamProducerHolder.create(proxyProvider, allocator);
+                }
+            } catch (Exception e) {
+                logger.error("Failed to create stream producer", e);
```

Comment:
The logging is inconsistent (sometimes `debug`, sometimes `error). we probably should use `debug` since `listener.error` will be logged upper in the call stack?
```suggestion
                logger.debug("Failed to create stream producer", e);
```

Reply:
i'm only logging error when some unexpected error occurs, otherwise its always debug. I think it might be useful to log these errors.